sky/provision/aws/instance.py

```diff
@@ -164,26 +164,29 @@ def _create_instances(ec2_fail_fast, cluster_name: str, node_config: Dict[str,
         'MaxCount': count,
         'TagSpecifications': tag_specs
     })
-
+    # We are adding 'NetworkInterfaces' in the inner loop and having both keys
+    # is considered invalid by the create_instances API.
+    security_group_ids = conf.pop('SecurityGroupIds', None)
     # NOTE: This ensures that we try ALL availability zones before
     # throwing an error.
     num_subnets = len(subnet_ids)
     max_tries = max(num_subnets * (BOTO_CREATE_MAX_RETRIES // num_subnets),
                     len(subnet_ids))
     per_subnet_tries = max_tries // num_subnets
+    assert 'NetworkInterfaces' not in conf, conf
```

Comment:
nit:
```suggestion
    assert conf.get('NetworkInterfaces') is None, conf
```
This is to allow a `NetworkInterfaces` specified to be `null`. Wdyt?

Reply:
We have already banned this key in the upper call path: 

https://github.com/skypilot-org/skypilot/blob/23aa6dd3f6daf49544d4ddd68acbce422c9d3d03/sky/provision/aws/config.py#L58-L61

Any reason we want to allow a null value?

Reply:
Oh, I see. I missed the upper level assertion. The current way sounds good to me then.